pkgs/development/libraries/physics/fastnlo_toolkit/default.nix

```diff
@@ -15,17 +16,23 @@
 
 stdenv.mkDerivation rec {
   pname = "fastnlo_toolkit";
-  version = "2.3.1pre-2411";
+  rev = "2823";
+  version = "2.5.0pre-${rev}";
 
-  src = fetchurl {
-    urls = [
-      "https://fastnlo.hepforge.org/code/v23/${pname}-${version}.tar.gz"
-      "https://sid.ethz.ch/debian/fastnlo/${pname}-${version}.tar.gz"
-    ];
-    sha256 = "0fm9k732pmi3prbicj2yaq815nmcjll95fagjqzf542ng3swpqnb";
-  };
+  src = fetchFromGitLab {
+    domain = "gitlab.etp.kit.edu";
+    owner = "qcd-public";
+    repo = "fastNLO";
+    inherit rev;
+    hash = "sha256-FEKnEnK90tT4BJJ6MLva9lCl3aYzO1YGdx/8Ol2vM7M=";
+  } + /v2.5/toolkit;
```

Comment:
```suggestion
  } + /v${lib.versions.majorMinor version}/toolkit;
```

Reply:
Still no

Reply:
Why do we want to hardcode a random version number?

Reply:
Because the whole path is arbitrary, there is no contract for it. Your suggestion provides a tiny benefit that is very well offset by how this is not human-readable and is a waste of people's times.

Reply:
Then why do we replace pname in URLs all over the place? You can't easily visit the URL and need to think extra hard if it is still correct.

Reply:
> Then why do we replace pname in URLs all over the place?

I don't think we should.



> Version numbers can be automatically updated

I don't think this package has automatic updating. If someone could make it work for this package, I'd say it's up to them to decide, knowing the technical capabilities of used tooling. Until that point, the discussion has no merit.

Reply:
> Then why do we replace pname in URLs all over the place?

Because some people were insisting on this a lot, so we ended up replacing with `${pname}` where possible, that was before automatic updates for `version` were a thing. I agree, it was pointless. In the rare case when project changes its name we can afford to spend time to edit the name in few places.

> You can't easily visit the URL and need to think extra hard if it is still correct.

So `${pname}` melts our brains, and the `/v${lib.versions.majorMinor version}/toolkit` melts them twice as hard :)

> Version numbers can be automatically updated and are the only thing that should be substituted everywhere.

The automatic updates are neat as a technology, but in practice I don't get much benefit from this as a maintainer. Not enough to entice me to go so far at replacing version strings. We have different priorities here.